protocol/communities/manager.go

```diff
@@ -4696,3 +4698,113 @@ func (m *Manager) createCommunityTokenPermission(request *requests.CreateCommuni
+		requestProto.RevealedAccounts = []*protobuf.RevealedAccount{}
+		syncRequestsWithoutRevealedAccounts = append(syncRequestsWithoutRevealedAccounts, requestProto)
+	}
+
+	syncMsgWithoutRevealedAccounts := &protobuf.CommunityPrivilegedUserSyncMessage{
+		Type:               protobuf.CommunityPrivilegedUserSyncMessage_CONTROL_NODE_ALL_SYNC_REQUESTS_TO_JOIN,
+		CommunityId:        community.ID(),
+		SyncRequestsToJoin: syncRequestsWithoutRevealedAccounts,
+	}
+
+	syncMsgWitRevealedAccounts := &protobuf.CommunityPrivilegedUserSyncMessage{
+		Type:               protobuf.CommunityPrivilegedUserSyncMessage_CONTROL_NODE_ALL_SYNC_REQUESTS_TO_JOIN,
+		CommunityId:        community.ID(),
+		SyncRequestsToJoin: syncRequestsWithRevealedAccounts,
+	}
+
+	subscriptionMsg := &CommunityPrivilegedMemberSyncMessage{
+		CommunityPrivateKey: community.PrivateKey(),
+	}
+
+	for role, members := range newPrivilegedMembers {
+		if len(members) == 0 {
+			continue
+		}
+
+		subscriptionMsg.Receivers = members
+
+		switch role {
+		case protobuf.CommunityMember_ROLE_ADMIN:
+			subscriptionMsg.CommunityPrivilegedUserSyncMessage = syncMsgWithoutRevealedAccounts
+		case protobuf.CommunityMember_ROLE_OWNER:
+			fallthrough
+		case protobuf.CommunityMember_ROLE_TOKEN_MASTER:
+			subscriptionMsg.CommunityPrivilegedUserSyncMessage = syncMsgWitRevealedAccounts
+		}
+
+		m.publish(&Subscription{CommunityPrivilegedMemberSyncMessage: subscriptionMsg})
+	}
+
+	return nil
+}
+
+func (m *Manager) shareAcceptedRequestToJoinWithPrivilegedMembers(community *Community, requestsToJoin *RequestToJoin) error {
+	pk, err := common.HexToPubkey(requestsToJoin.PublicKey)
+	if err != nil {
+		return err
+	}
+
+	acceptedRequestsToJoinWithoutRevealedAccounts := make(map[string]*protobuf.CommunityRequestToJoin)
+	acceptedRequestsToJoinWithRevealedAccounts := make(map[string]*protobuf.CommunityRequestToJoin)
+
+	acceptedRequestsToJoinWithRevealedAccounts[requestsToJoin.PublicKey] = requestsToJoin.ToCommunityRequestToJoinProtobuf()
+	requestsToJoin.RevealedAccounts = make([]*protobuf.RevealedAccount, 0)
+	acceptedRequestsToJoinWithoutRevealedAccounts[requestsToJoin.PublicKey] = requestsToJoin.ToCommunityRequestToJoinProtobuf()
+
+	msgWithRevealedAccounts := &protobuf.CommunityPrivilegedUserSyncMessage{
+		Type:          protobuf.CommunityPrivilegedUserSyncMessage_CONTROL_NODE_ACCEPT_REQUEST_TO_JOIN,
+		CommunityId:   community.ID(),
+		RequestToJoin: acceptedRequestsToJoinWithRevealedAccounts,
+	}
+
+	msgWithoutRevealedAccounts := &protobuf.CommunityPrivilegedUserSyncMessage{
+		Type:          protobuf.CommunityPrivilegedUserSyncMessage_CONTROL_NODE_ACCEPT_REQUEST_TO_JOIN,
+		CommunityId:   community.ID(),
+		RequestToJoin: acceptedRequestsToJoinWithoutRevealedAccounts,
+	}
+
+	// do not sent to control ourself and to the accepted user
+	skipMembers := make(map[*ecdsa.PublicKey]struct{})
+	skipMembers[&m.identity.PublicKey] = struct{}{}
```

Comment:
I think skipping ourselves might be a problem in case where we have an owner without the private key

Reply:
Here I'm not sure. By idea, we have a backup/sync mechanizm for sharing data between the devices

Reply:
ah you're right. This is just to sync, not to send a decision. Ignore my comment.